x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -178,6 +178,16 @@ public void roles(Set<String> roleNames, FieldPermissionsCache fieldPermissionsC
         }
     }
 
+    public void getRoleDescriptors(Set<String> roleNames, ActionListener<Set<RoleDescriptor>> listener) {
+        roleDescriptors(roleNames, ActionListener.wrap(rolesRetrievalResult -> {
+            if (rolesRetrievalResult.isSuccess()) {
+                listener.onResponse(rolesRetrievalResult.getRoleDescriptors());
+            } else {
+                listener.onResponse(Collections.emptySet());
```

Comment:
This will silently create api-keys with no roles if rolesRetrieval failed. Shouldn't that be a `failure` ?